fedora_bot.py

```diff
@@ -317,24 +317,27 @@ def main():
             parser.error(f"Invalid component format, must be PACKAGE:NUM_TESTS : {component_numtests}")
 
         print(f"\n--- {component} ---\n")
-        if args.apikey:
-            msg_info(f"Checking for open pull requests of {component}...")
-            merge_open_pull_requests(args, component, num_tests)
-        else:
-            msg_info("No Fedora account API key supplied - skipping merging of pull requests.")
-
-        if args.user and args.password: # Only check Bodhi if credentials were supplied
-            msg_info(f"Checking for missing updates of '{component}'...")
-            missing_updates = get_missing_updates(component, fedoras)
-
-            if missing_updates:
-                msg_info(f"Found missing updates in Bodhi: {missing_updates}")
-                publish_updates(args, component, missing_updates)
-                msg_ok(f"Tried to update {missing_updates}.")
+        try:
+            if args.apikey:
+                msg_info(f"Checking for open pull requests of {component}...")
+                merge_open_pull_requests(args, component, num_tests)
             else:
-                msg_ok("No releases found with missing updates.")
-        else:
-            msg_info("No Fedora credentials supplied - skipping Bodhi updates.")
+                msg_info("No Fedora account API key supplied - skipping merging of pull requests.")
+
+            if args.user and args.password: # Only check Bodhi if credentials were supplied
+                msg_info(f"Checking for missing updates of '{component}'...")
+                missing_updates = get_missing_updates(component, fedoras)
+
+                if missing_updates:
+                    msg_info(f"Found missing updates in Bodhi: {missing_updates}")
+                    publish_updates(args, component, missing_updates)
+                    msg_ok(f"Tried to update {missing_updates}.")
+                else:
+                    msg_ok("No releases found with missing updates.")
+            else:
+                msg_info("No Fedora credentials supplied - skipping Bodhi updates.")
+        except:
+            print(f"Failure in processing component [{component}] - skipping")
```

Comment:
Is there a specific exception that we could be catching? In addition, I think that it would be good to actually print the exception message, potentially with the traceback as part of the message. Otherwise we are loosing potentially useful information, which would make debugging harder.

Reply:
I have updated the code, hope it looks better now :)

Reply:
My proposition would be to evaluate this code next time it will fail and start adding detected exceptions one by one (and eventually handle them specified ways).